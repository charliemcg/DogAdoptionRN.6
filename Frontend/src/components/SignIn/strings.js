export default {
  email: 'Email',
  password: 'Password',
  signIn: 'Sign In',
  username: 'Username',
  userApi: 'http://127.0.0.1:8000/end_user/end_user/',
  applicationJson: 'application/json',
  undefined: 'undefined',
  userNotExist: 'ERROR! User does not exist',
  navigation: {
    home: 'Home',
  },
};
