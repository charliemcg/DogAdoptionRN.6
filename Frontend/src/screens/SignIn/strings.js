export default {
  signInFaceBook: "Sign in with Facebook",
  newMember: "New member?",
  signUp: "Sign Up!",
  navigation: {
    underConstruction: "UnderConstruction",
    signUp: "SignUp"
  }
};
