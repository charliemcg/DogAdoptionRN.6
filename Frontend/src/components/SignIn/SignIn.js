/**
 * User Profile Notes
 *
 * Users can sign up via the sign up page.
 * Existing users can sign in via the sign in page.
 * Non existing users cannot sign in via the sign in page.
 * User details are stored in redux only while signed in.
 *
 * TODO
 * Encrypt passwords
 * Allow users to enter their first name, last name and location
 * Allow users to change their details
 * User picker instead of CharField for location
 * Make it so that email address works as primary key and get rid of usernames
 * Implement character limits on user details
 * Implement password constraints
 *
 */

import React, {Component} from 'react';
import {connect} from 'react-redux';
import {
  View,
  Text,
  TouchableHighlight,
  TextInput,
  KeyboardAvoidingView,
} from 'react-native';
import IconAwesome from 'react-native-vector-icons/FontAwesome5';
import IconMCI from 'react-native-vector-icons/MaterialCommunityIcons';
import styles from './styles';
import colors from '../../colors';
import {signInOut, setUser} from '../../actions';
import strings from './strings';

class SignIn extends Component {
  constructor(props) {
    super(props);
    this.state = {
      user: {
        first_name: 'Test',
        last_name: 'User',
        username: '',
        location: 'SA',
        password: '',
      },
    };
  }

  onChangeUsername(text) {
    this.setState({
      user: {
        first_name: this.state.user.first_name,
        last_name: this.state.user.last_name,
        username: text,
        location: this.state.user.location,
        password: this.state.user.password,
      },
    });
  }

  onChangePassword(text) {
    this.setState({
      user: {
        first_name: this.state.user.first_name,
        last_name: this.state.user.last_name,
        username: this.state.user.username,
        location: this.state.user.location,
        password: text,
      },
    });
  }

  updateBackend() {
    // figure out if signing up or signing in
    if (this.props.signUp) {
      fetch('http://127.0.0.1:8000/end_user/end_user/', {
        method: 'POST',
        headers: {
          Accept: 'application/json',
          'Content-Type': 'application/json',
        },
        body: JSON.stringify(this.state.user),
      })
        .then(() => {
          this.props.setUser(this.state.user);
        })
        .then(() => {
          this.props.signInOut();
          this.props.navigation.navigate(strings.navigation.home);
        })
        .catch(e => console.log(e));
    } else {
      fetch(
        `http://127.0.0.1:8000/end_user/end_user/${this.state.user.username}/`,
      )
        .then(resp => {
          return resp.json();
        })
        .then(json => {
          //reject users which do not already exist
          if (typeof json.username === 'undefined') {
            throw 'User does not exist';
          } else {
            user = {
              first_name: json.first_name,
              last_name: json.last_name,
              username: json.username,
              location: json.location,
              password: json.password,
            };
            this.props.setUser(user);
          }
        })
        .then(() => {
          this.props.signInOut();
          this.props.navigation.navigate(strings.navigation.home);
        })
        .catch(e => console.log(`ERROR! ${e}`));
    }
  }

  render() {
    return (
      <KeyboardAvoidingView style={{flex: 3}} behavior="padding" enabled>
        {/* username */}
        <View style={styles.usernameWrapper}>
          <View style={styles.textInput}>
            <IconAwesome
              name="user"
              size={30}
              color={colors.primary}
              style={styles.inputIcon}
            />
            {/* textContentType is for autofill  */}
            <TextInput
              textContentType="username"
              placeholder={strings.username}
              style={styles.inputText}
              onChangeText={text => this.onChangeUsername(text)}
            />
          </View>
        </View>
        {/* password */}
        <View style={styles.passwordWrapper}>
          <View style={styles.textInput}>
            <IconMCI
              name="textbox-password"
              size={30}
              color={colors.primary}
              style={styles.inputIcon}
            />
            {/* textContentType is for autofill use newPassword for sign up page */}
            <TextInput
              textContentType="password"
              placeholder={strings.password}
              secureTextEntry={true}
              style={styles.inputText}
              onChangeText={text => this.onChangePassword(text)}
            />
          </View>
        </View>
        {/* sign in */}
        <View style={styles.signInWrapper}>
          <TouchableHighlight
            style={styles.signIn}
            onPress={() => {
              this.updateBackend();
            }}
            underlayColor={colors.dark}>
            <Text style={styles.signInText}>{strings.signIn}</Text>
          </TouchableHighlight>
        </View>
      </KeyboardAvoidingView>
    );
  }
}

const mapDispatchToProps = dispatch => {
  return {
    signInOut: () => {
      dispatch(signInOut());
    },
    setUser: user => {
      dispatch(setUser(user));
    },
  };
};

export default connect(null, mapDispatchToProps)(SignIn);
