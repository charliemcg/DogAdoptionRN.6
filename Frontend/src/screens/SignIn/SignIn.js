import React, {Component} from 'react';
import {
  View,
  Text,
  TouchableHighlight,
  TextInput,
  ImageBackground,
} from 'react-native';
import styles from './styles';
import IconAwesome from 'react-native-vector-icons/FontAwesome5';
import IconMCI from 'react-native-vector-icons/MaterialCommunityIcons';
import colors from '../../colors';
import strings from './strings';
import {signInOut} from '../../actions';
import {connect} from 'react-redux';
import backgroundImg from '../../images/backgroundWhite.png';
import SignInComponent from '../../components/SignIn';
import Device from 'react-native-device-detection';

class SignIn extends Component {
  render() {
    return (
      <ImageBackground
        style={styles.parent}
        resizeMode="cover"
        source={backgroundImg}>
        {/* icon */}
        <View style={styles.iconWrapper}>
          <IconAwesome
            name="dog"
            size={Device.isTablet ? 150 : 100}
            color={colors.primary}
          />
        </View>
        {/* sign in component */}
        <SignInComponent navigation={this.props.navigation} signUp={false} />
        {/* sign in with Facebook */}
        <View style={styles.facebookWrapper}>
          <TouchableHighlight
            style={styles.facebook}
            onPress={() =>
              this.props.navigation.navigate(
                strings.navigation.underConstruction,
              )
            }>
            <Text style={styles.facebookText}>{strings.signInFaceBook}</Text>
          </TouchableHighlight>
        </View>
        {/* new user link */}
        <View style={styles.privacyWrapper}>
          <Text>{strings.newMember}</Text>
          <Text
            style={{color: colors.dark}}
            onPress={() =>
              this.props.navigation.navigate(strings.navigation.signUp)
            }>
            {strings.signUp}
          </Text>
        </View>
      </ImageBackground>
    );
  }
}

export default SignIn;
