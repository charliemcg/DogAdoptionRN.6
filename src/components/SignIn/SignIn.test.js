import "react-native";
import React from "react";
import SignIn from "./SignIn";
import renderer from "react-test-renderer";
import store from "../../store";

test("SignIn snapShot", () => {
  const snap = renderer.create(<SignIn store={store} />).toJSON();

  expect(snap).toMatchSnapshot();
});
