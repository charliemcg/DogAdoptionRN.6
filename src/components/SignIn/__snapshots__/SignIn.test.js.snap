// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn snapShot 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flex": 3,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "width": 750,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "#ffd54f",
          "borderBottomStyle": "solid",
          "borderBottomWidth": 2,
          "flexDirection": "row",
          "height": 75,
          "justifyContent": "center",
          "width": 375,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#ffd54f",
              "fontSize": 30,
            },
            Object {
              "flex": 1,
            },
            Object {
              "fontFamily": "FontAwesome5Free-Regular",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {
              "fontWeight": "400",
            },
          ]
        }
      >
        
      </Text>
      <TextInput
        allowFontScaling={true}
        placeholder="Email"
        rejectResponderTermination={true}
        style={
          Object {
            "flex": 5,
            "fontSize": 18,
            "height": "100%",
          }
        }
        textContentType="username"
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "width": 750,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "#ffd54f",
          "borderBottomStyle": "solid",
          "borderBottomWidth": 2,
          "flexDirection": "row",
          "height": 75,
          "justifyContent": "center",
          "width": 375,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#ffd54f",
              "fontSize": 30,
            },
            Object {
              "flex": 1,
            },
            Object {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <TextInput
        allowFontScaling={true}
        placeholder="Password"
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Object {
            "flex": 5,
            "fontSize": 18,
            "height": "100%",
          }
        }
        textContentType="password"
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "width": 750,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffd54f",
          "borderRadius": 10,
          "height": 75,
          "justifyContent": "center",
          "width": 249.75,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#37474f",
            "fontSize": 20,
          }
        }
      >
        Sign In
      </Text>
    </View>
  </View>
</View>
`;
